internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -1003,39 +993,48 @@ func (s *Server) processPullRequestEvent(
 		return nil, fmt.Errorf("error creating pull request properties: %w", err)
 	}
 
-	provider, err := s.providerManager.InstantiateFromID(ctx, repoEnt.Entity.ProviderID)
+	repoProps, err := properties.NewProperties(map[string]any{
+		properties.PropertyUpstreamID: strconv.FormatInt(ghRepo.GetID(), 10),
+		properties.PropertyName:       ghRepo.GetName(),
+	})
 	if err != nil {
-		return nil, fmt.Errorf("error instantiating provider: %w", err)
+		return nil, fmt.Errorf("error creating repository properties for PR origination: %w", err)
 	}
 
-	prEntWithProps, err := s.reconcilePrWithDb(ctx, provider, repoEnt, pullProps)
-	if errors.Is(err, errNotHandled) {
-		return nil, err
-	} else if err != nil {
-		return nil, fmt.Errorf("error reconciling PR with DB: %w", err)
+	// it is bit of a code smell to use the fetcher here just to format the name
+	fetcher := ghprop.NewPullRequestFetcher()
+	name, err := fetcher.GetName(pullProps)
```

Comment:
can you call this in-line so you don't need to allocate memory?

e.g.

```
name, err := ghprop.PullRequestFetcher{}.GetName(pullProps)
```

or something of the sort

Reply:
Memory for `fetcher` is going to be allocated regardless of the pattern, but this way is more readable.
Note that fields within `PullRequestFetcher` are global variables, so their allocation is already payed for.

Reply:
done (in a separate patch for easier re-review)